### PR TITLE
 Add invite button to profile

### PR DESCRIPTION
**Pull request by markulrich**

### CODE REVIEW

File: src/components/RequestInvite.js [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    this.setState({ videoUrl: event.target.value })
+  }
+
+  setFullName = async (event: any): void => {
+    this.setState({ fullName: event.target.value });
+  }
+
+  handleOnSubmit = async (event: any): void => {
+    event.stopPropagation();
+    if (!getYoutubeUrlVideoId(this.state.videoUrl)) {
```
Comment (ataki) [inline src/components/RequestInvite.js]: the best experience, coming from the user perspective, is to handle validation during `onblur` event of the input.

can make a pr for that once we merge this

File: src/components/Profile.js [hunk 2]
```diff
@@ -70,20 +51,30 @@ class Profile extends Component<ProfileProps, ProfileState> {
...
+  return <div>Loading</div>;  // TODO make work
 };
 
-export default Profile;
+function mapStateToProps(state, ownProps) {
+  const isMyProfile = ownProps.match.path === '/me';
+  const memberId = isMyProfile ? null : ownProps.match.params.memberId;
+  const authIsLoaded = state.auth.isLoaded;
+  const authFirebaseUser = state.auth.firebaseUser;
+  const authMemberDoc = getAuthMemberDoc(state);  // TODO will we use this?
```
Comment (ataki) [inline src/components/Profile.js]: observation after working with the data fetching layer: there's 2 (potentially more) ways of getting the current authorized user: (1) importing the `auth` module, and (2) fetching the user by memberId. Juxtaposed with methods to get another member's doc, can be confusing.

is there a way we could unify the two modules?

File: src/components/RequestInvite.js [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+  }
+  const authMember = state.members.byUid[authFirebaseUser.uid];
+  return {
+    isAuthLoaded,
+    authFirebaseUser,
+    memberId,
+    isAuthMemberDocLoaded: authMember && !authMember.isFetching,
+    authMemberDoc: getAuthMemberDoc(state),
+    fullName: state.auth.firebaseUser.displayName,
+    toMemberDoc: getMemberDocByMid(state, ownProps.match.params.memberId)
```
Comment (ataki) [inline src/components/RequestInvite.js]: could we use `memberId` here instead? e.g. `getMemberDocByMid(state, memberId)` or do they refer to different things?

File: src/components/RequestInvite.js [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+      <div>
+        <b>Request Invite</b>
+        {this.renderDirections()}
+        {this.props.notSignedIn ? this.renderLogIn() : (this.props.isAuthLoaded ? this.renderForm() : <div>Loading</div>)}
+      </div >
+    );
+  }
+}
+
+function mapStateToProps(state, ownProps) {
```
Comment (ataki) [inline src/components/RequestInvite.js]: just so I understand when `fetchMemberByUidIfNeeded` is dispatched, this logic will rerun to update the props of the view, correct?

**Comment by ataki** [comment]
actually, might want to wait until after the internationalization is fixed to merge; it's easier to incorporate changes from that branch into this one than the other way around

**Comment by markulrich** [comment]
@ataki @kem226 what's the timeline for merging in i18n? We currently have no translations available so it's fairly low priority, no rush in my book

File: src/components/Profile.js (hunk 2 above)
Comment (markulrich) [reply src/components/Profile.js]: sounds like a good idea!

File: src/components/RequestInvite.js (hunk 3 above)
Comment (markulrich) [reply src/components/RequestInvite.js]: The memberDoc is firebase doc where we can do `.get('full_name')` etc ... we should get Flow actually working and be better about documenting our types (perhaps with a linter), we've probably now gone a bit too far from "full typescript" to "barely typed"

File: src/components/RequestInvite.js (hunk 4 above)
Comment (markulrich) [reply src/components/RequestInvite.js]: Yep! I think `mapStateToProps` will be called whenever the redux state changes, and if a fetch is needed then `fetchMemberByUidIfNeeded` will cause redux state change

**Pull request merged**